src/Text/Parsing/StringParser.purs

```diff
@@ -26,26 +26,25 @@ type Pos = Int
 type PosString = { str :: String, pos :: Pos }
 
 -- | The type of parsing errors.
-newtype ParseError = ParseError String
-
-instance showParseError :: Show ParseError where
-  show (ParseError msg) = msg
-
-derive instance eqParseError :: Eq ParseError
-
-derive instance ordParseError :: Ord ParseError
+type ParseError = { error :: String, pos :: Pos }
 
 -- | A parser is represented as a function which takes a pair of
 -- | continuations for failure and success.
-newtype Parser a = Parser (PosString -> Either { pos :: Pos, error :: ParseError } { result :: a, suffix :: PosString })
+newtype Parser a = Parser (PosString -> Either ParseError { result :: a, suffix :: PosString })
```

Comment:
We could potentially create a type alias for this, too:

```suggestion
type ParseResult a = { result :: a, suffix :: PosString }

newtype Parser a = Parser (PosString -> Either ParseError (ParseResult a))
```

Reply:
Why?

Reply:
The same record is used in several places, so this would just reduce a bit of duplication. It's not necessary, though.